### PR TITLE
test: support hover state for shape element

### PR DESCRIPTION
**Pull request by zqran**
tests for #1282 , #1248

### CODE REVIEW

**Comment by zqran** [comment]
I will add another tests.

File: tests/edgeless.spec.ts [hunk 1]
```diff
@@ -288,3 +288,56 @@ test('selection box of shape element sync on fast dragging', async ({
...
+  await switchMouseMode(page);
+
+  await page.mouse.move(150, 150);
+
+  const hoverRect = page.locator('.affine-edgeless-hover-rect');
+  const box = await hoverRect.boundingBox();
+  if (box === null) throw new Error();
+
+  const { x, y, width, height } = box;
+  expect([x, y, width, height]).toStrictEqual([100, 100, 100, 100]);
```
Comment (doodlewind) [inline tests/edgeless.spec.ts:308]: Can we have an `assertHoverRect` util for this?

Comment (zqran) [reply tests/edgeless.spec.ts:308]: ok, i'll change it

File: tests/edgeless.spec.ts [hunk 2]
```diff
@@ -282,9 +283,54 @@ test('selection box of shape element sync on fast dragging', async ({
...
+  await switchMouseMode(page);
+  await dragBetweenCoords(page, { x: 100, y: 100 }, { x: 200, y: 200 });
+  await switchMouseMode(page);
+
+  await page.mouse.move(150, 150);
+
+  const hoverRect = page.locator('.affine-edgeless-hover-rect');
+  const box = await hoverRect.boundingBox();
+
+  assertHoverRect(box, ['x', 'y', 'width', 'height'], [100, 100, 100, 100]);
```
Comment (zqran) [inline tests/edgeless.spec.ts:303]: Is this okay?

Comment (doodlewind) [reply tests/edgeless.spec.ts:303]: I think `assertHoverRect(page, box, 100, 100, 100, 100)` should make more sense? And this util should also handle the `page.locator('.affine-edgeless-hover-rect')` setup.

Comment (zqran) [reply tests/edgeless.spec.ts:303]: In this mode `assertHoverRect(page, box, 100, 100, 100, 100)`, `box` is a selector for `page.locator(box)`?

Comment (zqran) [reply tests/edgeless.spec.ts]: if this function just for 'hover rect', we can use this `assertHoverRect(page, 100, 100, 100, 100)`

Comment (doodlewind) [reply tests/edgeless.spec.ts]: done :)

**Pull request merged**